cmd/entrypoint/testutil_fake_syntheticauth_test.go

```diff
@@ -0,0 +1,119 @@
+package entrypoint_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/datawire/ambassador/v2/cmd/entrypoint"
+	v3bootstrap "github.com/datawire/ambassador/v2/pkg/api/envoy/config/bootstrap/v3"
+	v3cluster "github.com/datawire/ambassador/v2/pkg/api/envoy/config/cluster/v3"
+	"github.com/datawire/ambassador/v2/pkg/snapshot/v1"
+)
+
+func HasAuthService(namespace, name string) func(snapshot *snapshot.Snapshot) bool {
+	return func(snapshot *snapshot.Snapshot) bool {
+		for _, m := range snapshot.Kubernetes.AuthServices {
+			if m.Namespace == namespace && m.Name == name {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+// Tests the synthetic auth generation when a valid AuthService is created
+// This authservice has protocol_Version: v3 and should not be replaced by the synthetic AuthService
+func TestSyntheticAuthWithValid(t *testing.T) {
+	t.Setenv("EDGE_STACK", "true")
+
+	f := entrypoint.RunFake(t, entrypoint.FakeConfig{EnvoyConfig: true}, nil)
+
+	err := f.UpsertYAML(`
+---
+apiVersion: getambassador.io/v3alpha1
+kind: AuthService
+metadata:
+  name: edge-stack-auth-test
+  namespace: foo
+spec:
+  auth_service: 127.0.0.1:8500
+  protocol_version: "v3"
+  proto: "grpc"
+`)
+	assert.NoError(t, err)
+	f.Flush()
+
+	snap, err := f.GetSnapshot(HasAuthService("foo", "edge-stack-auth-test"))
+	assert.NoError(t, err)
+	assert.NotNil(t, snap)
+
+	assert.Equal(t, "edge-stack-auth-test", snap.Kubernetes.AuthServices[0].Name)
+
+	// Check for an ext_authz cluster name matching the provided AuthService (Http_Filters are harder to check since they always have the same name)
+	isAuthCluster := func(c *v3cluster.Cluster) bool {
+		return strings.Contains(c.Name, "cluster_extauth_127_0_0_1_8500_foo")
+	}
+
+	// Grab the next envoy config that satisfies our predicate.
```

Comment:
And very much here -- it makes me sad that this "Grab the next envoy config that satisfies our predicate" comment gets copied around so much, because it doesn't really help. 😢 

"Grab the next Envoy config that has an Edge Stack auth cluster on 127.0.0.1:8500" maybe?